### PR TITLE
Fix JPEG data

### PR DESCRIPTION
**Pull request by valentinperignon**
Keep creation and modification date

### CODE REVIEW

File: kDriveCore/Utils/PHAsset+Exension.swift [hunk 1]
```diff
@@ -96,14 +95,27 @@ extension PHAsset {
...
+    private func getJpegData(for resource: PHAssetResource, options: PHAssetResourceRequestOptions) async throws -> Data? {
         return try await withCheckedThrowingContinuation { continuation in
             var imageData = Data()
-            PHAssetResourceManager.default().requestData(for: resource, options: requestResourceOption) { data in
+            PHAssetResourceManager.default().requestData(for: resource, options: options) { data in
                 // Get all pieces of data
                 imageData.append(data)
             } completionHandler: { error in
-                if let error = error {
+                if let error {
```
Comment (adrien-coye) [inline kDriveCore/Utils/PHAsset+Exension.swift:118]: That's cool we no longer need the `error = error`

File: kDriveCore/Utils/PHAsset+Exension.swift [hunk 2]
```diff
@@ -80,8 +80,7 @@ extension PHAsset {
         let targetURL = FileImportHelper.instance.generateImportURL(for: resourceUTI)
         do {
             if shouldTransformIntoJPEG {
```
Comment (adrien-coye) [inline kDriveCore/Utils/PHAsset+Exension.swift:82]: I think we can use guards to achieve the same with less nested ifs

```
do {
            guard shouldTransformIntoJPEG == true else {
                try await PHAssetResourceManager.default().writeData(for: resource, toFile: targetURL, options: requestResourceOption)
                return targetURL
            }
            
            guard let jpegData = try await getJpegData(for: resource, requestResourceOption: requestResourceOption) else {
                return nil
            }
            
            try jpegData.write(to: targetURL)
            return targetURL
        }
```

File: kDriveCore/Utils/PHAsset+Exension.swift [hunk 3]
```diff
@@ -96,14 +95,27 @@ extension PHAsset {
         return nil
     }
 
-    private func getJpegData(for resource: PHAssetResource, requestResourceOption: PHAssetResourceRequestOptions) async throws -> Data? {
+    private func writeJpegData(to url: URL, resource: PHAssetResource, options: PHAssetResourceRequestOptions) async throws -> Bool {
+        if let jpegData = try await getJpegData(for: resource, options: options) {
```
Comment (adrien-coye) [inline kDriveCore/Utils/PHAsset+Exension.swift:99]: We can use a guard here also I think

**Review by adrien-coye — APPROVED** [review]
All good 👍

**Review by PhilippeWeidmann — APPROVED** [review]

**Pull request merged**